### ISSUE TITLE
Suboptimal suggestion from implied_bounds_in_impls when adding bounds to associated type

### ISSUE THREAD

**Issue by progval** [rust-lang/rust-clippy#13617]
### Summary

When returning a type with redundant bounds on an associated type, `implied_bounds_in_impls`'s suggestion duplicates the implied bound

### Reproducer

I tried this code ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=16dcb13696a5881cb663b8c6a31f843c)]:

```rust
<pub trait MyTrait {
    type MyAssociatedType;
}

pub trait MyStricterTrait: MyTrait<MyAssociatedType: Clone> {}

pub struct MyStruct;
pub struct MyStruct2;

impl MyTrait for MyStruct {
    type MyAssociatedType = u64;
}

impl MyTrait for MyStruct2 {
    type MyAssociatedType = Vec<u64>;
}

impl MyStricterTrait for MyStruct {
}

pub fn f() -> impl MyStricterTrait + MyTrait<MyAssociatedType: Copy> {
    MyStruct
}>
```

I expected to see this happen:

```
warning: this bound is already specified as the supertrait of `MyStricterTrait`
  --> src/lib.rs:21:38
   |
21 | pub fn f() -> impl MyStricterTrait + MyTrait<MyAssociatedType: Copy> {
   |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#implied_bounds_in_impls
   = note: `#[warn(clippy::implied_bounds_in_impls)]` on by default
help: try removing this bound
   |
21 - pub fn f() -> impl MyStricterTrait + MyTrait<MyAssociatedType: Copy> {
21 + pub fn f() -> impl MyStricterTrait<MyAssociatedType: Copy> {
   |
```

Instead, this happened:

```
warning: this bound is already specified as the supertrait of `MyStricterTrait`
  --> src/lib.rs:21:38
   |
21 | pub fn f() -> impl MyStricterTrait + MyTrait<MyAssociatedType: Copy> {
   |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#implied_bounds_in_impls
   = note: `#[warn(clippy::implied_bounds_in_impls)]` on by default
help: try removing this bound
   |
21 - pub fn f() -> impl MyStricterTrait + MyTrait<MyAssociatedType: Copy> {
21 + pub fn f() -> impl MyStricterTrait<MyAssociatedType: Copy> + MyTrait<MyAssociatedType: Copy> {
   |
```

### Version

default versions on the Rust playground (rustc 1.82.0)
